packages/test/test-end-to-end-tests/src/test/simpleLoadTest.spec.ts

```diff
@@ -0,0 +1,93 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { ContainerRuntimeFactoryWithDefaultDataStore, DataObject, DataObjectFactory } from "@fluidframework/aqueduct";
+import {
+    IContainer,
+    IRuntimeFactory,
+} from "@fluidframework/container-definitions";
+import { Container } from "@fluidframework/container-loader";
+import {
+    IContainerRuntimeOptions,
+} from "@fluidframework/container-runtime";
+import { requestFluidObject } from "@fluidframework/runtime-utils";
+import { ITestObjectProvider } from "@fluidframework/test-utils";
+import { describeFullCompat } from "@fluidframework/test-version-utils";
+import { flattenRuntimeOptions } from "./flattenRuntimeOptions";
+
+class TestDataObject extends DataObject {
+    public get _root() {
+        return this.root;
+    }
+
+    public get _context() {
+        return this.context;
+    }
+}
+
+describeFullCompat("Simple load test", (getTestObjectProvider) => {
+    let provider: ITestObjectProvider;
+    const factory = new DataObjectFactory(
+        "TestDataObject",
+        TestDataObject,
+        [],
+        []);
+
+    const runtimeOptions: IContainerRuntimeOptions = {
+        summaryOptions: {
+            generateSummaries: false,
+        },
+        gcOptions: {
+            gcAllowed: true,
+        },
+    };
+    const defaultRuntimeFactory = new ContainerRuntimeFactoryWithDefaultDataStore(
+        factory,
+        [
+            [factory.type, Promise.resolve(factory)],
+        ],
+        undefined,
+        undefined,
+        flattenRuntimeOptions(runtimeOptions),
+    );
+
+    let container: IContainer;
+
+    const createContainer = async (runtimeFactory: IRuntimeFactory): Promise<IContainer> => {
+        return provider.createContainer(runtimeFactory);
+    };
+
+    async function ensureContainerConnected(c: IContainer): Promise<void> {
+        if (!(c as Container).connected) {
+            return new Promise((resolve) => c.once("connected", () => resolve()));
```

Comment:
Everybody does that. Should we refactor RunWhileConnectedCoordinator.watiStart() into more useful method and stop writing this kind of code? We can come up with more generic interface to abstract away differences between runtime & container

Reply:
We might want to add some generic mechanism for this since this is a very common check. We can add something to LoaderContainerTracker too since most of the tests use it.